listenbrainz/db/stats.py

```diff
@@ -40,24 +42,18 @@ def get_timestamp_for_last_user_stats_update():
         return row['last_update_ts'] if row else None
 
 
-def insert_user_stats(user_id, artists, recordings, releases):
+def insert_user_stats(user_id: int, artists: UserArtistStatJson, recordings: dict, releases: dict):
```

Comment:
We should consider refactoring this into three different functions, this does not seem ideal to me.

Reply:
at the moment different types of stats are in different columns though - any thoughts on this? Would you have a single column with a dict of artist/recording/release counts, or separate tables, or a type column? Do you expect to calculate all types of stats for a user at once, or a single type of stats for all users at once?

Reply:
A single type of stat for all users, this has been refactored by @ishaanshah  already.